internal/resources/cluster/resource_cluster.go

```diff
@@ -250,10 +271,10 @@ func resourceClusterCreate(ctx context.Context, d *schema.ResourceData, m interf
 	// always run
 	d.SetId(clusterResponse.Cluster.Meta.UID)
 
-	if kubeconfigfile != "" {
+	if _, ok := d.GetOk(attachClusterKey); ok {
```

Comment:
This condition functionality will fail when attachClusterKey is defined but none of the attributes are added inside the block

Reply:
For that terraform will error out from schema as expected min items is 1.

Reply:
That block has description , so if you just provide description and not other values how will ur code work ?

Reply:
Is the description really needed as a parameter to be passed, I just see it defined in the schema and not used any where else in the code base, is it something that will be largely used by the provider users?

Reply:
I am not talking about the use case here.
Pointing to the the code and the corner case which you are missing

Reply:
Description inside the attach k8s might be used by the user to describe something which would help them identify the kubeconfig which is being used.

Reply:
Added schema validation which will ensure that one of the ways of defining kubeconfig have been added and it will throw validation error from terraform to handle this case.

Reply:
Here, we should be checking is k8s client is nil or not

Reply:
Added the check

Reply:
Added